### ISSUE TITLE
Add ability to rerun a failed build with the same resource versions

### ISSUE THREAD

**Issue by ry4n1m3** [concourse/concourse#129]
We have a pipeline A -> B. A produces a resource X. If B consumes X and fails, we'd like to (optionally) rerun B with the same X. In our environment A runs super fast, while B takes forever, so its possible A has run 1-3 times in the time it took B to fail.

To complicate this further, we have this 4 times in parallel, which can make moving a particular git revision through our pipeline very difficult (lots of manual babysitting).

- Ops Manager Team

**Comment by vito** [comment]
closing in favor of https://github.com/concourse/concourse/issues/413 which is a bit more up-to-date
